### PR TITLE
Add ref assemblies property

### PR DESCRIPTION
**Pull request by gewarren**
Follow up to #28057. 

[Preview](https://review.docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props?branch=pr-en-us-28059#producereferenceassemblyinoutdir).

### CODE REVIEW

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -266,6 +266,18 @@ The `PreserveCompilationReferences` property is similar to the [PreserveCompilat
 
 For more information, see [Razor SDK properties](/aspnet/core/razor-pages/sdk#properties).
 
+### ProduceReferenceAssemblyInOutDir
+
+The `ProduceReferenceAssemblyInOutDir` property controls whether reference assemblies are written to the `OutDir` directory. Staring in .NET 6, the default value is `false`, and reference assemblies are only written to the `IntermediateOutputPath` directory. Set the value to `true` to write reference assemblies to the `OutDir` directory.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:271]:
```suggestion
The `ProduceReferenceAssemblyInOutDir` property controls whether reference assemblies are written to the `OutDir` directory. Starting in .NET 6, the default value is `false`, and reference assemblies are only written to the `IntermediateOutputPath` directory. Set the value to `true` to write reference assemblies to the `OutDir` directory.
```

Comment (gewarren) [reply docs/core/project-sdk/msbuild-props.md:271]: Was this property only introduced in .NET 6?

Comment (baronfel) [reply docs/core/project-sdk/msbuild-props.md:271]: Seems like! I jumped into the msbuild code history and it was added ~2 months ago in the same commit that the ref assemblies change was made.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

**Pull request merged**